mimic/test/test_swift.py

```diff
@@ -119,74 +313,617 @@ def test_get_container(self):
                 b"X-Container-Bytes-Used")[0], b"0"
         )
 
-    def test_get_no_container(self):
+    def test_get_container_non_existent(self):
         """
         GETing a container that has not been created results in a 404.
         """
-        self.createSwiftService()
         # create a container
-        uri = (self.json_body['access']['serviceCatalog'][0]['endpoints'][0]
-               ['publicURL'] + '/testcontainer').encode("ascii")
-        container_response = self.successResultOf(
-            request(self, self.root, b"GET", uri)
+        container_response, _ = self.get_container(
+            expected_result=404, with_body=False)
+
+        # Validate the response
+        self.assertIsNone(
+            container_response.headers.getRawHeaders(
+                "X-Container-Object-Count")
         )
-        self.assertEqual(container_response.code, 404)
+        self.assertIsNone(
+            container_response.headers.getRawHeaders(
+                "X-Container-Bytes-Used")
+        )
+
+    def test_get_container_non_empty(self):
+        """
+        Test retrieving a non-empty container
```

Comment:
For future reference, please avoid using terms like "test" and "validate" in test docstrings - https://jml.io/pages/test-docstrings.html has a good primer.